### ISSUE TITLE
Possible daylight savings time bug in test?

### ISSUE THREAD

**Issue by fryfrog** [jeffaco/duplicacy-util#31]
I'm writing an AUR PKGBUILD for this project and one of the tests fails w/ an hour difference that seems suspiciously like daylight savings time issue.

```
=== RUN   TestTimeDiffStringWithDurationDays
--- FAIL: TestTimeDiffStringWithDurationDays (0.00s)
    timeutils_test.go:250: Result was incorrect, got '4 days, 2:02:05', expected '4 days, 3:02:05'.
```

What else can I provide to be helpful?

**Comment by jeffaco** [comment]
I’m out of town right now, returning this weekend. So expect delayed responses until I get home.

It’s odd that I don’t see this locally (pacific time zone, normally), nor do I see it on Travis-CI (which is likely UTC, but I’d need to check).

What time zone are you running in? If, for test purposes, you change to Pacific Timezone (Los Angeles timezone), does the problem persist or go away?

**Comment by fryfrog** [comment]
No rush, obviously! I'm actually PST too, I'll try to dig deeper. I'm totally new to Go, so it'll be slow. :)

**Comment by jeffaco** [comment]
In my case, if I just install `go`, download `duplicate-util`, install dependencies as per the README file, and do a `go test`, it passes just fine.

You mentioned that you’re trying to build in AUR PKGBUILD. I have no familiarity with that at all. If you build as I do (or as Travis-CI does) and do `go test` directly, do the unit tests pass or fail then?

**Comment by fryfrog** [comment]
Even when done manually, it has the same issue. I've set `$GOPATH` to `/home/fryfrog/go` and cleaned that folder out, then re-ran all the `go get` commands. It builds fine and all the tests work except that one for time.

```
0 ✓ fryfrog@apollo ~ $ go get github.com/djherbis/times
0 ✓ fryfrog@apollo ~ $ go get github.com/mitchellh/go-homedir
0 ✓ fryfrog@apollo ~ $ go get github.com/spf13/viper
0 ✓ fryfrog@apollo ~ $ go get github.com/theckman/go-flock
0 ✓ fryfrog@apollo ~ $ go get gopkg.in/gomail.v2
0 ✓ fryfrog@apollo ~ $ cd go/src/
0 ✓ fryfrog@apollo ~/go/src $ ls
github.com  golang.org  gopkg.in
0 ✓ fryfrog@apollo ~/go/src $ git clone https://github.com/jeffaco/duplicacy-util.git
Cloning into 'duplicacy-util'...
remote: Enumerating objects: 14, done.
remote: Counting objects: 100% (14/14), done.
remote: Compressing objects: 100% (12/12), done.
remote: Total 441 (delta 3), reused 7 (delta 2), pack-reused 427
Receiving objects: 100% (441/441), 15.25 MiB | 17.99 MiB/s, done.
Resolving deltas: 100% (260/260), done.
0 ✓ fryfrog@apollo ~/go/src $ cd duplicacy-util/
0 ✓ fryfrog@apollo ~/go/src/duplicacy-util $[master] go build
0 ✓ fryfrog@apollo ~/go/src/duplicacy-util $[master] go test
--- FAIL: TestTimeDiffStringWithDurationDays (0.00s)
    timeutils_test.go:250: Result was incorrect, got '4 days, 2:02:05', expected '4 days, 3:02:05'.
FAIL
exit status 1
FAIL    duplicacy-util  1.191s
1 ✗ fryfrog@apollo ~/go/src/duplicacy-util $[master] echo $GOPATH
/home/fryfrog/go
0 ✓ fryfrog@apollo ~/go/src/duplicacy-util $[master] readlink /etc/localtime
/usr/share/zoneinfo/America/Los_Angeles
0 ✓ fryfrog@apollo ~/go/src/duplicacy-util $[master] date
Fri Nov  2 13:33:52 PDT 2018
```

**Comment by jeffaco** [comment]
Okay, I'm going to see if I can repro this. Looks like you're running this on Linux, yes? What vendor/version?

I noted that Go v1.11.2 came out. I don't expect it to help, but can you upgrade to that, just in case?

**Comment by jeffaco** [comment]
Forget it, I understand the issue. Like you said, it's a daylight savings time issue. Last night, when I realized daylight savings time was coming up, a bunch of tests were failing. This morning, all pass.

Now that I can replicate it, I can look at this.

**Comment by fryfrog** [comment]
Great, and I was using Arch Linux, not that it matters now. :)

**Comment by jeffaco** [comment]
Fixed in commit hash cbb98b1. Closing issue. Thanks for reporting this!
